api/envoy/admin/v2alpha/config_dump.proto

```diff
@@ -178,3 +183,38 @@ message RoutesConfigDump {
   // The dynamically loaded route configs.
   repeated DynamicRouteConfig dynamic_route_configs = 3 [(gogoproto.nullable) = false];
 }
+
+// This message describes the control plane configuration that Envoy is currently connected to for
+// various xDS implementations.
+message ControlPlaneConfigDump {
+
+  // This message describes the control plane information per each config source.
+  message ConfigSourceControlPlaneInfo {
+
+    // gRPC service represented by ApiConfigSource.
+    envoy.api.v2.core.GrpcService grpc_service = 1;
+
+    // The control plane that this Service is currently connected to.
+    envoy.api.v2.core.ControlPlane control_plane = 2;
+
+    // The timestamp when the ControlPlane information was last updated.
+    google.protobuf.Timestamp last_updated = 3;
+  }
+
+  // This is message describes the control plane configuration for each xDS service that Envoy
+  // connects to.
+  message ServiceControlPlaneInfo {
+
+    // The xDS service this control plane information belongs to. For example
+    // envoy.api.v2.EndpointDiscoveryService represents control plane info that Envoy has been
+    // configured for EndpointDiscoveryService.
+    string server_type_url = 1;
```

Comment:
Actually, this isn't type URL, it's the fully qualified service name, so maybe call it `xds_service`. Sorry about the confusion.